packages/affiliates/test/mocha/attribution-rewards.js

```diff
@@ -0,0 +1,21 @@
+const Path = require("path");
+const assert = require("assert");
+
+const { DappMining } = require("../../libs/affiliates");
+const { getWeb3 } = require("@uma/common");
+const { getAbi } = require("@uma/core");
+const { mocks } = require("../../libs/datasets");
+const { Queries } = mocks;
+
+const web3 = getWeb3();
+const empAbi = getAbi("ExpiringMultiParty");
+const datasetPath = Path.join(__dirname, "../datasets/set1");
+
+describe("DappMining", function() {
+  it("should instanciate", function() {
+    const queries = Queries(datasetPath);
+    const result = DappMining({ queries, empAbi, web3 });
+    assert(result);
+  });
+  // TODO: add more tests and appropriate dataset
```

Comment:
this feels like it's going to be a pretty epic task to add tests...

Reply:
yea..... question is should we add tests before we merge? I think we should have tests before we actually execute this script in production

Reply:
will be working on this #2365